src/clients/SpokePoolClient/SpokePoolClient.ts

```diff
@@ -960,17 +753,75 @@ export class SpokePoolClient extends BaseAbstractClient {
+   * and store them in memory. This method is the primary method for updating the state of this client.
+   * @param eventsToQuery An optional list of events to query. If not provided, all events will be queried.
+   * @returns A Promise that resolves to a SpokePoolUpdate object.
+   */
+  protected abstract _update(eventsToQuery: string[]): Promise<SpokePoolUpdate>;
+
+  /**
+   * Finds the deposit id at a specific block number.
+   * @param blockTag The block number to search for the deposit ID at.
+   * @returns The deposit ID.
+   */
+  public abstract _getDepositIdAtBlock(blockTag: number): Promise<BigNumber>;
+
   /**
-   * Find the amount filled for a deposit at a particular block.
-   * @param relayData Deposit information that is used to complete a fill.
-   * @param blockTag Block tag (numeric or "latest") to query at.
-   * @returns The amount filled for the specified deposit at the requested block (or latest).
+   * @notice Return maximum of fill deadline buffer at start and end of block range. This is a contract
+   * immutable state variable so we can't query other events to find its updates.
+   * @dev V3 deposits have a fill deadline which can be set to a maximum of fillDeadlineBuffer + deposit.block.timestamp.
+   * Therefore, we cannot evaluate a block range for expired deposits if the spoke pool client doesn't return us
+   * deposits whose block.timestamp is within fillDeadlineBuffer of the end block time. As a conservative check,
+   * we verify that the time between the end block timestamp and the first timestamp queried by the
+   * spoke pool client is greater than the maximum of the fill deadline buffers at the start and end of the block
+   * range. We assume the fill deadline buffer wasn't changed more than once within a bundle.
+   * @param startBlock start block
+   * @param endBlock end block
+   * @returns maximum of fill deadline buffer at start and end block
    */
-  public relayFillStatus(
+  public abstract getMaxFillDeadlineInRange(startBlock: number, endBlock: number): Promise<number>;
+
+  /**
+   * Retrieves the time from the SpokePool contract at a particular block.
+   * @returns The time at the specified block tag.
+   */
+  public abstract getTimeAt(blockNumber: number): Promise<number>;
+
+  /**
+   * For a given origin chain depositId, resolve the corresponding Deposit.
+   * Note: This method can only be used for depositIds within the non-deterministic range (0 < depositId < 2^32 - 1).
+   * @param depositId Deposit ID of the deposit to resolve.
+   * @returns A DepositSearchResult instance.
+   */
+  public abstract findDeposit(depositId: BigNumber): Promise<DepositSearchResult>;
+
+  /**
+   * Retrieves the timestamp for a given block number.
+   * @param blockTag The block number to retrieve the timestamp for.
+   * @returns The timestamp for the given block number.
+   */
+  public abstract getTimestampForBlock(blockTag: number): Promise<number>;
+
```

Comment:
Same as above - we can fortunately drop this.

```suggestion
```

Reply:
https://github.com/across-protocol/sdk/pull/935